packages/botfuel-dialog/src/bot.js

```diff
@@ -143,25 +145,64 @@ class Bot {
     }
   }
 
+  /**
+   * Handles a user message.
+   * @async
+   * @param {Object} userMessage - the user extended message
+   * @returns {Promise.<void>}
+   */
+  async handleMessage(userMessage) {
+    let botMessages = [];
+
+    logger.debug('handleMessage', userMessage);
+
+    const contextIn = {
+      user: userMessage.user,
+      brain: this.brain,
+      userMessage,
+      config: this.config,
+    };
+
+    await this.middlewareManager.in(contextIn, async () => {
+      botMessages = await this.respond(userMessage);
+    });
+
+    const contextOut = {
+      user: userMessage.user,
+      brain: this.brain,
+      botMessages,
+      config: this.config,
+      userMessage,
+    };
+    await this.middlewareManager.out(contextOut, async () => {});
+
+    return botMessages;
+  }
+
   /**
    * Responds to the user.
    * @async
    * @param {Object} userMessage - the user message
    * @returns {Promise.<void>}
    */
   async respond(userMessage) {
+    let botMessages;
+
     logger.debug('respond', userMessage);
```

Comment:
Maybe this can be the first instruction.

Reply:
done